File: 0801-0900/0822-Card Flipping Game/0822-Card Flipping Game.cpp
class Solution {
public:
    int flipgame(vector<int>& fronts, vector<int>& backs) {
        int n = fronts.size();
        unordered_set<int> table;
        for (int i = 0; i < n; ++i) {
            if (fronts[i] == backs[i]) {
                table.insert(fronts[i]);
            }
        }
        
        int output = INT_MAX;
        for (int i = 0; i < n; ++i) {
            if (!table.count(fronts[i])) {
                output = min(output, fronts[i]);
            }
            if (!table.count(backs[i])) {
                output = min(output, backs[i]);
            }
        }
        return output == INT_MAX ? 0 : output;
    }
};

File: 0801-0900/0822-Card Flipping Game/0822-Card Flipping Game.py
import itertools
class Solution:
    def flipgame(self, fronts: List[int], backs: List[int]) -> int:
        same = {x for x, y in zip(fronts, backs) if x == y}
        output = 2001
        for x in itertools.chain(fronts, backs):
            if x not in same:
                output = min(output, x)
        return output % 2001
